packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -219,40 +221,45 @@ realm_on_dictionary_change_func_t get_on_dictionary_change() {
 }
 
 realm_notification_token_t *
-register_notification_cb(int64_t collection_ptr, realm_collection_type_e collection_type,
-                         jobject callback) {
+register_notification_cb(
+        int64_t collection_ptr,
+        realm_collection_type_e collection_type,
+        int64_t key_path_array_ptr,
+        jobject callback
+) {
     auto user_data = static_cast<jobject>(get_env()->NewGlobalRef(callback));
     auto user_data_free = [](void *userdata) {
         get_env(true)->DeleteGlobalRef(static_cast<jobject>(userdata));
     };
 
     switch (collection_type) {
-        case RLM_COLLECTION_TYPE_NONE: return realm_object_add_notification_callback(
+        case RLM_COLLECTION_TYPE_NONE:
+            return realm_object_add_notification_callback(
                     reinterpret_cast<realm_object_t*>(collection_ptr),
                     user_data, // Use the callback as user data
                     user_data_free,
-                    NULL, // See https://github.com/realm/realm-kotlin/issues/661
+                    (key_path_array_ptr == 0) ? NULL : reinterpret_cast<realm_key_path_array_t*>(key_path_array_ptr),
```

Comment:
Is this nullability check required?

Reply:
Strictly no, but it makes it easier to understand what is going on IMO